liferay-frontend/liferay-portal
# LPS-115436 Closes the Dropdown Support utility when clicking outside the Dropdown

## Test Case

1. Create 5 or more Web Content articles to get the pagination toolbar to appear within the Web Content admin page
2. Within the pagination toolbar, click on the items per page dropdown menu to expand the list of values
3. Click anywhere outside of the expanded dropdown menu

Now, the dropdown can be closed when clicking outside the dropdown panel.

## Questions

Hey @wincent, @jbalsas, We have a lot of function for detecting if a click target was clicked outside the component or not. See https://gist.github.com/diegonvs/8d858111311f0fec05b4a55bb9a9e962

We have the following code patterns:

```javascript
	/**
			 * Detect clicks on elements or their descendent elements.
			 */
			const filtered = elements.filter(
				(element) => element && element.contains(event.target)
			);

			if (!filtered.length) {
				handler(event);
			}
```

and

```javascript
export default (target, ...elements) => {
	return !elements.some((element) => {
		if (typeof element === 'string') {
			return !!dom.closest(target, element);
		}

		return element && dom.contains(element, target);
	});
};
```

I prefer the first impl since We can avoid using metal-dom.

Should We consider moving `isClickOutside`(please, suggest a better naming if possible) to `frontend-js-web`?

Also, Fixes https://issues.liferay.com/browse/LPS-108106
